impedance/circuits.py

```diff
@@ -26,7 +26,10 @@ def __init__(self, initial_guess, constants={}, name=None, bounds=None):
 
         # initalize class attributes
         self.initial_guess = initial_guess
-        self.constants = constants
+        if constants is not None:
```

Comment:
Using:
if constants:

here would allow you to leave it as an empty dictionary and still tell if it's empty. Empty lists, tuples, and dictionaries all return false if they're empty when used in that context.

Reply:
This actually doesn't look great to me either, but I ended up removing the empty dictionary because apparently its not great practice to set a mutable default argument (e.g., https://stackoverflow.com/a/9526480) and it was flagging it as an issue in codacy :smile: